### DISCUSSION TITLE
Qustion about the `LINUX_VERSION` argument

### DISCUSSION THREAD

**Discussion by infdahai** [openzfs/zfs#14888] [Q&A]
I want to learn zfs so I compile and build it in WSL distro. But I exec the `modprobe zfs` failed.

The issue information is throwed in https://github.com/microsoft/WSL/issues/10128. In short, I find  the build and `extra` dir depends on the `LINUX_VERSION` argument.

The [kernel.m4](https://github.com/openzfs/zfs/blob/ad0a554614b096698d9969340c4c593690042d5b/config/kernel.m4#LL477C18-L477C28) generate this argument. But I don't really understand the configuration.
```m4
LINUX_VERSION=${kernsrcver}
```

And the `makefile` file shows this.

```makefile
LINUX = /usr/src/linux-headers-5.15.0-72-generic
LINUX_OBJ = /usr/src/linux-headers-5.15.0-72-generic
LINUX_SYMBOLS = Module.symvers
LINUX_VERSION = 5.15.0-72-generic
```

The log from the command of`make install` is shown below.

```
make -C /usr/src/linux-headers-5.15.0-72-generic M="$PWD" modules_install \
	INSTALL_MOD_PATH= \
	INSTALL_MOD_DIR=extra \
	KERNELRELEASE=5.15.0-72-generic
make[2]: Entering directory '/usr/src/linux-headers-5.15.0-72-generic'
  INSTALL /lib/modules/5.15.0-72-generic/extra/spl.ko
  SIGN    /lib/modules/5.15.0-72-generic/extra/spl.ko
  INSTALL /lib/modules/5.15.0-72-generic/extra/zfs.ko
  SIGN    /lib/modules/5.15.0-72-generic/extra/zfs.ko
  DEPMOD  /lib/modules/5.15.0-72-generic
```

So I want to ask how this argument corresponds to `5.15.0-72-generic` and fix this issue from `WSL`.

**Comment by rincebrain** [comment]
The problem is you need headers somewhere that match your kernel version, not that it's picking the wrong version - for reasons I don't understand, people here seem to think picking a random kernel header that seems vaguely similar in version if it can't find the right one is a feature.

**Reply by infdahai** [reply]
yeah, you are right. I didn't think about it
